lib/cardano-api-extra/lib/Cardano/Api/Gen.hs

```diff
@@ -1760,6 +1760,19 @@ genVotingProcedures w = case w of
             , pure TxVotingProceduresNone
             ]
 
+genSupplementalData
+    :: CardanoEra era
+    -> ( Gen
+            (BuildTxWith BuildTx (ShelleyApi.TxSupplementalDatums era2))
+       )
+genSupplementalData era = fmap BuildTxWith
+    $ flip (inEonForEra (pure ShelleyApi.TxSupplementalDataNone)) era
+    $ \case
+        ConwayEraOnwardsConway -> oneof
+            [ pure ShelleyApi.TxSupplementalDataNone
+            , pure $ ShelleyApi.TxSupplementalDatums []
```

Comment:
Have not tried to compile this, but:
```suggestion
            , ShelleyApi.TxSupplementalDatums <$> genHashableScriptData
```
Otherwise a TODO would be nice given the `[]` and generator name.

Reply:
fixed